### ISSUE TITLE
feature: multiple send-receive target-directories

### ISSUE THREAD

**Issue by Geremia** [digint/btrbk#214]
Instead of having in my `btrbk.conf`, e.g.
```
target send-receive    /mnt/dest1
target send-receive    /mnt/dest2
```
it would be good if by specifying
```
target send-receive    /mnt/dest1    /mnt/dest2
```
`btrbk` would use `btrfs-send` to send to dest1 and dest2 simultaneously, i.e., do something like this (cf. "[How do I send btrfs snapshots to multiple destination drives?](https://stackoverflow.com/q/48234297/1429450)"):
```
btrfs send /tmp/incremental_backup_snapshot | tee >(btrfs receive /mnt/destination_drive2) | btrfs receive /mnt/destination_drive1
```
This would be useful for making mirrored backups on multiple drives.

**Comment by digint** [comment]
Mmh, this is a nice idea, but it would probably cause some non-trivial changes in btrbk: When doing incremental backups, btrbk looks for the "latest common parent" on both sides. With this proposal, there will be three (or ultimately, N) sides to compare, introducing problems like: What if a optimal parent is present on the first target, but not on the second? Should we use the second-best parent then? Or revert to sending twice?.

**Comment by Gatak** [comment]
" What if a optimal parent is present on the first target, but not on the second? Should we use the second-best parent then? Or revert to sending twice?."

I like the idea of having either as options in the config file.

Another potential issue is what to do if one target fails

A third option might be to use a temporary middle storage for the btrfs-send and use that temp file to send to multiple targets,perhaps in parallell. This would save performance on the live source subvol.

**Comment by GuillaumeSeren** [comment]
Hey,
I think it could be done by adding an option on the btrbk call to filter the target,
and use several services in parallel, providing the same feature,
without adding the multi-target complexity in the script.

**Comment by bdaase** [comment]
Would it maybe already be enough to send the snapshots to all, individually, specified targets in parallel instead of one by one?
This would significantly improve throughout.

**Comment by schildbach** [comment]
Yes, parallelizing the entire process (and separating into one lock per target) is an interesting idea. However, network congestion because of multiple backups over the same slow line, and disk trashing needs to be taken into account.
